lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionSourceCode.cpp

```diff
@@ -64,58 +130,126 @@ struct CallsAndArgs {
 /// - And a matching call to the sink function:
 ///
 /// lldb_sink($__lldb_arg, $__lldb_injected_self, $τ_0_0, $τ_0_1, ..., $τ_0_n)
-static CallsAndArgs MakeGenericSignaturesAndCalls(
-    llvm::ArrayRef<SwiftASTManipulator::VariableInfo> local_variables) {
+static llvm::Expected<CallsAndArgs> MakeGenericSignaturesAndCalls(
```

Comment:
This functions seems to be doing quite a lot now, and a lot of the new behavior depends on whether `generic_sig` exists and if `needs_object_ptr` is true/false,  I wonder if it'd be possible to split it up in maybe 2,3 functions